submissions/api.py

```diff
@@ -447,27 +447,27 @@ def get_all_course_submission_information(course_id, item_type, read_replica=Tru
     if read_replica:
         submission_qs = _use_read_replica(submission_qs)
 
-    query = submission_qs.select_related('student_item').prefetch_related('score_set').filter(
+    query = submission_qs.select_related('student_item__scoresummary__latest__submission').filter(
```

Comment:
This `select_related` means that, for each `submission` in our query, we are fetching:
- its student_item, which can be tied to a
- scoresummary, which can be tied to a
- score (latest), which can be tied to a
- submission

The upshot is that, for each submission, we can ask the question `If we tried to get the score for this submission's student_item, would the score reported be associated with this submission?` If it is, the score is included in our data download. If it is not, no score is returned.

This is to handle a situation like this one:
- student makes a submission
- student gets a poor score on that submission
- student makes a new submission
- student gets a better score on that submission
The returned values in this case would be both submissions, but no score would be reported for the first one. Is this desirable? Or would it be better to return both scores even though one of them doesn't count? We have elected to go this route for now, since that seems to be how the api is set up for `get_score`.

Worth noting: ORA imposes a limit of 1 submission per student item (excluding delete state requests, but that data already excluded), so this is irrelevant to the context of ORA data downloads. It's more a question of how edx-submissions want to handle this theoretical case for other consumers in the future.